src/compiler/checker.ts

```diff
@@ -7347,11 +7347,13 @@ namespace ts {
         function getContextualTypeForBinaryOperand(node: Expression): Type {
             const binaryExpression = <BinaryExpression>node.parent;
             const operator = binaryExpression.operatorToken.kind;
-            if (operator >= SyntaxKind.FirstAssignment && operator <= SyntaxKind.LastAssignment) {
+
+            if (SyntaxKind.FirstAssignment <= operator && operator <= SyntaxKind.LastAssignment) {

```

Comment:
Purely stylistic, but everywhere else in the compiler we always put the constant on the right hand side (as the code was originally written). I'd prefer to keep things consistent.